src/main/java/org/openhab/tools/analysis/checkstyle/readme/MarkdownCheck.java

```diff
@@ -94,23 +93,54 @@ private void checkReadMe(FileText fileText) {
         Set<Class<? extends Block>> enabledBlockTypes = new HashSet<>(
                 Arrays.asList(Heading.class, ListBlock.class, FencedCodeBlock.class, IndentedCodeBlock.class));
         Node readmeMarkdownNode = parseMarkdown(fileText, enabledBlockTypes);
+
         // CallBack is used in order to use the protected methods of the AbstractStaticCheck in the Visitor
         MarkdownVisitorCallback callBack = new MarkdownVisitorCallback() {
             @Override
-            public int findLineNumber(FileText fileContent, String searchedText, int startLineNumber)
-                    throws NoResultException {
-                return MarkdownCheck.this.findLineNumber(fileText, searchedText, startLineNumber);
+            public int findLineNumber(FileText fileContent, String searchedText, int startLineNumber,
+                    LineFormatterCallback lineFormatterCallback) throws NoResultException {
+                return MarkdownCheck.this.findLineNumberFormatted(fileText, searchedText, startLineNumber,
+                        lineFormatterCallback);
             }
 
             @Override
             public void log(int line, String message) {
                 MarkdownCheck.this.log(line, message);
             }
         };
-        MarkdownVisitor visitor = new MarkdownVisitor(callBack, fileText);
+        MarkdownVisitor visitor = new MarkdownVisitor(callBack, fileText, new LineFormatterCallback() {
```

Comment:
If the design stays as is, I would make the `LineFormatterCallback` a [`FunctionalInterface`](http://www.baeldung.com/java-8-functional-interfaces)

Reply:
Great suggestion, I will use it.